examples/3_simple_storage.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env python3
+
+###################################
+# WHAT IS IN THIS EXAMPLE?
+#
+# Keybase has added an encrypted key-value store intended to support
+# security-conscious bot development with persistent state. It is a place to
+# store small bits of data that are
+#   (1) encrypted for a team or user (via the user's implicit self-team: e.g.
+# alice,alice),
+#   (2) persistent across logins
+#   (3) fast and durable.
+#
+# It supports putting, getting, listing, and deleting. There is also concurrency
+# support, but check out 5_secret_storage.py for that. A team has many
+# namespaces, a namespace has many entryKeys, and an entryKey has one current
+# entryValue. Namespaces and entryKeys are in cleartext, and the Keybase client
+# service will encrypt and sign the entryValue on the way in (as well as
+# decrypt and verify on the way out) so keybase servers cannot see it or forge
+# it.
+#
+# This example shows how you can use KVStoreClient to interact with the team
+# encrypted key-value store.
+###################################
+
+import asyncio
+import logging
+import os
+import sys
+from enum import Enum
+
+from pykeybasebot import Bot, EventType
+
+logging.basicConfig(level=logging.DEBUG)
+
+if "win32" in sys.platform:
+    # Windows specific event-loop policy
+    asyncio.set_event_loop_policy(
+        asyncio.WindowsProactorEventLoopPolicy()  # type: ignore
+    )
+
+
+class KVMsg(Enum):
+    PUT = "put"
+    GET = "get"
+    DELETE = "delete"
+    LIST = "list"
+    HELP = "help"
+
+
+class KVHandler:
+    """
+    KVHandler handles commands sent via chat to use the team key-value store.
+
+    KVHandler listens to chat messages of the form:
+    `!storage put <namespace> <key> <value> (<revision>)`
+    `!storage get <namespace> <key>`
+    `!storage delete <namespace> <key> (<revision>)`
+    `!storage list`  // list namespaces
+    `!storage list <namespace>`  // list entries in namespace
+    """
+
+    MSG_PREFIX = "!storage"
+
+    async def __call__(self, bot, event):
+        members_type = event.msg.channel.members_type
+        if not event.type == EventType.CHAT:
+            return
+
+        channel = event.msg.channel
+        user = event.msg.sender.username
+
+        # support teams and implicit self teams
+        team = channel.name
+        if members_type == "impteamnative" and channel.name == user:
+            team = "{0},{0}".format(channel.name)
```

Comment:
not an issue for this PR, but we should think about if there's a way we can push this logic down into the kvstore_client. or better, even deeper into the service so we can just remove this completely.

Reply:
https://keybase.atlassian.net/browse/PICNIC-641